python/tank/api.py

```diff
@@ -798,8 +799,76 @@ def preview_filesystem_structure(self, entity_type, entity_id, engine=None):
     """
-    Creates a Toolkit Core API instance based on a path inside a project
-    or path pointing directly at a pipeline configuration.
+    Creates a Toolkit Core API instance based on a path to a configuration
+    or a path to any file inside a project root location.
+
+    This factory method will do the following:
+
+    - If the given path is determined to be pointing at a pipeline configuration,
+      checks will be made to determine that the currently imported ``sgtk`` module is
+      associated with the configuration. There are two ways this can be valid:
+
+        - Either the configuration is localized, meaning that has it's
+          own ``sgtk`` core module. In that case, it is validated that the ``sgtk``
+          module currently running is exactly that core API belonging to the configuration.
+
+        - ...or the configuration is sharing an ``sgtk`` core with other configurations.
+          in this case, is is validated that the currently imported ``sgtk`` module
+          is that shared core that the configuration requires.
+
+    - If the given path is to a file (e.g. a maya file for example), the method will
+      do the following:
+
+       - Retrieve all projects from Shotgun, including project their `Project.tank_name`
+         project root folder fields, and associated pipeline configurations.
+
+       - Walk up the path hierarchy of the given path until one of the project roots are
+         matching the path.
+
+       - Get a list of matching pipeline configurations for that project.
+
+       - Ensure that the currently import ``sgtk`` module is a valid configuration for
+         the matching configurations. If more than one configuration is matching, the
+         primary will take precendence.
+
+    This method can be used if you have a toolkit project installed in a particular location and
+    want to write a script that initializes it. For example, if a Toolkit configuration
+    is set up in ``/mnt/projects/hidden_forest/tk_config``, you could initialize it like this::
+
+        # add the core of the project to the pythonpath
+        import sys
+        sys.path.append("/mnt/projects/hidden_forest/tk_config/install/core/python")
+
+        # now import the API
+        import sgtk
+
+        # import the configuration
+        tk = sgtk.sgtk_from_path("/mnt/projects/hidden_forest/tk_config/install/core/python")
+
+    If you have a shared core for all your projects, you could pass a file path to this
+    method and toolkit would automatically figure out which configuration would be suitable::
+
+        # add the shared core to the pythonpath
+        import sys
+        sys.path.append("/mnt/toolkit/shared_core")
+
+        # now import the API
+        import sgtk
+
+        # request that the API produced a tk instance suitable for a given file
+        tk = sgtk.sgtk_from_path("/mnt/projects/hidden_forest/shots/aa/aa_001/lighting/foreground.v002.ma")
+
+    .. note::
```

Comment:
Wonder if this note should be at the top? That would push people toward the manager before reading through the `sgtk_from_path` specifics.